server/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java

```diff
@@ -1336,4 +1336,41 @@ public void test_alias_in_left_join_condition() {
         execute(stmt);
         assertThat(response).hasRows("2| bazinga");
     }
+
+    /**
+     * https://github.com/crate/crate/issues/13503
+     */
+    @Test
+    public void test_nested_joins() {
+        execute("CREATE TABLE j1 (x INT)");
+        execute("CREATE TABLE j2 (x INT)");
+        execute("CREATE TABLE j3 (x INT)");
+
+        execute(
+            """
+                SELECT *
+                  FROM j1
+                  JOIN (j2 JOIN j3 ON j2.x = j3.x)
+                   ON j1.x = j2.x
+                """
+        );
+
+        assertThat(response).hasRows("");
```

Comment:
Maybe worth asserting the explain output too or adding some rows to ensure there's a sensible plan?